src/main/groovy/com/brightsparklabs/gradle/baseline/BaselinePlugin.groovy

```diff
@@ -165,22 +166,68 @@ public class BaselinePlugin implements Plugin<Project> {
     private void setupDependencyLicenseReport(final Project project) {
         project.plugins.apply "com.github.jk1.dependency-license-report"
 
-        // Register the temp file manager and the config override tasks
-        project.tasks.register('bslCheckLicenseManageTempFile', CheckLicenseManageTempFileTask)
-        project.tasks.register('bslOverrideAllowedLicenses', OverrideAllowedLicensesTask)
+        def tmpBaselineDir = new File("${project.buildDir}/tmp/brightsparklabs/baseline/")
+        def overrideBaselineDir = new File("${project.projectDir}/brightsparklabs/baseline/")
+
+        /*
+            Creates an override file with the default configuration from the internal baseline
+            allowed-licenses.json file
+        */
+        project.task("bslOverrideAllowedLicenses") {
+            group = "brightSPARK Labs - Baseline"
+            description = "Creates an override file for the config of the types of allowed licenses that " +
+                    "dependencies can have. This config file is used by the `checkLicense` task."
+
+            outputs.file("${overrideBaselineDir}/allowed-licenses.json")
+
+            doLast {
+                if (!overrideBaselineDir.exists()) {
+                    overrideBaselineDir.mkdirs()
+                }
+
+                outputs.files.singleFile.text = getClass().getResourceAsStream("/allowed-licenses.json").getText()
+            }
+        }
+
+        /*
+            The checkLicense task requires a java.io.File type as an input, this task manages the creation of this input
+            file as a temp file in the build directory. It uses the configuration of the baseline allowed-licenses.json
+            or an override configuration file from running the bslOverrideAllowedLicenses task.
+        */
+        project.task("bslGenerateAllowedLicenses") {
+            group = "brightSPARK Labs - Baseline"
+            description = "Generates the config file for the `checkLicense` task using either " +
+                    "the default baseline config or a supplied config override file."
+
+            inputs.files("${overrideBaselineDir}/allowed-licenses.json").optional()
+            outputs.file("${tmpBaselineDir}/allowed-licenses.json")
+
+            doLast {
+                String allowedLicensesConfig
+                def jsonSlurper = new JsonSlurper()
+
+                if (inputs.files.singleFile.exists()) {
+                    allowedLicensesConfig = JsonOutput.toJson(jsonSlurper.parseText(inputs.files.singleFile.text))
+                } else {
+                    allowedLicensesConfig = getClass().getResourceAsStream("/allowed-licenses.json").getText()
+                }
+
+                if (outputs.files.singleFile.exists()) {
+                    if (outputs.files.singleFile.text == allowedLicensesConfig) {
+                       return
```

Comment:
Not really worth the check, it doesn't matter if we overwrite the file with the same text
The output caching should have hopefully caught this